dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -898,20 +916,20 @@ public void removeAttachedFile(WorkflowTask task, String fileInode) throws DotDa
 			return Collections.emptyList();
 		}
         final List<WorkflowAction> actions = workFlowFactory.findActions(step);
-        return filterActionsCollection(actions, user, true, permissionable);
+        return permissionsHelper.filterActions(actions, user, RESPECT_FRONTEND_ROLES, permissionable);
     }
 
 	@Override
 	@CloseDBIfOpened
-	public List<WorkflowAction> findActions(final WorkflowStep step, final Role role) throws DotDataException,
+	public List<WorkflowAction> findActions(final WorkflowStep step, final Role role, final Permissionable permissionable) throws DotDataException,

```

Comment:
I am wondering if you shouldn't keep the previous method and call the new one passing null as a permissionable